### PR TITLE
Add kriging to rastergrid of pointdata from shapefile using pykrige 

### PR DESCRIPTION
**Pull request by Christian-Mielke**
This is handy for generating raster grids to be compared in a GIS with e.g. TROPOMI raster data later on.
You need to have GDAL and pykrige installed to make it work.

### CODE REVIEW

File: pyscript/shape2raster.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import geopandas as gpd
+import numpy
+import os
+from osgeo import gdal
+from scipy import interpolate
+import pykrige as pykg
+
+
+#  class for writing raster data
```
Comment (kevin-hausmann) [inline pyscript/shape2raster.py:9]: Please use docstrings instead

File: pyscript/shape2raster.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+class Rasterfun(object):
+    def __init__(self):
```
Comment (kevin-hausmann) [inline pyscript/shape2raster.py:11]: This is not needed

File: pyscript/shape2raster.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import geopandas as gpd
+import numpy
+import os
+from osgeo import gdal
+from scipy import interpolate
+import pykrige as pykg
+
+
+#  class for writing raster data
+class Rasterfun(object):
```
Comment (kevin-hausmann) [inline pyscript/shape2raster.py:10]: No need to inherit from object, this is fine (besides the class name)
```suggestion
class Rasterfun:
```

File: pyscript/shape2raster.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        dimat = numpy.shape(numpyarray)
+        dataset_out = driver.Create(out, dimat[2], dimat[1], dimat[0], gdal.GDT_Float32)
+        count = 1
+        while count <= dimat[0]:
+            dim = count - 1
+            dataset_out.GetRasterBand(count).WriteArray(numpyarray[dim, :, :])
+            count += 1
+        dataset_out = None  # Close dataset clean
+
+    def schreibebsqsingle(numpyarray: numpy.ndarray, out: str):
```
Comment (kevin-hausmann) [inline pyscript/shape2raster.py:30]: Needs to be static or have parameter `self`, see all the above.

**Review by kevin-hausmann — CHANGES_REQUESTED** [review]
Hi there, unfortunately there are a couple of issues with this PR, including:
- The purpose of this whole endeavour is not clearly explained
- Deletes a bunch of unrelated files (incl. the result.csv)
- Add a local .gitignore instead of using the global one
- There is no documentation on this classes and methods
- There are no unit tests
- There is German/English language mixing
- Use of classes is unneeded as there is no internal state
- Class names such as "Rasterfun" are non-descriptive

I also put some more comments inline below, but stopped as those would be too many. Sorry, be this needs reworking. I'd be happy to help you with this, we can surely find the time to do some pair programming...

**Comment by AnnaZagorski** [comment]
then I will add my review after the rework

**Comment by kevin-hausmann** [comment]
@Christian-Mielke Will you still be working on this? Is this still relevant? Otherwise, please remove and delete the branch, thanks!

**Pull request closed without merging**